src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt

```diff
@@ -29,6 +34,34 @@ object MovementListener : Listener {
         }
     }
 
+    @EventHandler(ignoreCancelled = true)
+    fun onPlayerFalling(playerMoveEvent: PlayerMoveEvent) {
+        if(DeeperConfig.data.maxFallingDistance == -1f){
+            return
+        }
+
+        val (player) = playerMoveEvent
+        val actualFallDistance = player.getVehicleRecursive()?.fallDistance ?: player.fallDistance
+        if (actualFallDistance > DeeperConfig.data.maxFallingDistance
+                && !player.isGliding
+                && (player.gameMode == GameMode.SURVIVAL || player.gameMode == GameMode.ADVENTURE)) {
+            val previousTick = previousFallingDamageTick[player.uniqueId]
+
+            if (previousTick != null && currentServerTick - previousTick < player.maximumNoDamageTicks) {
+                return
+            }
+
+            // Always deal a minimum of 1 damage, else the first damage tick would deal (almost) no damage
+            val damageToDeal = (actualFallDistance - DeeperConfig.data.maxFallingDistance) * DeeperConfig.data.fallingDamageMultiplier + 1
```

Comment:
Wouldn't this be negative (i.e. healing player) if `actualFallDistance < maxFallingDistance`? I'm assuming max fall distance is the maximum safe falling distance before it starts hurting you, is that right?

If this is the case, it's probably better to `damageToDeal.coerceIn(0.0, player.health)`, plus you avoid having to get the max health through an attribute and can safely do `player.health -= damageToDeal`.

Reply:
Thats why I have a check to make sure that the actualFallDistance is larger than maxFallingDistance before running this section of code. I like your suggestion though in order to remove the property retrieval, I'll do that.

Reply:
Ah sorry you're right, I just didn't notice that above hehe

Reply:
Just a quick note, I'll keep this mostly how it was. If `player.health` is assigned a value that makes it negative, it throws an exception. Also since I want the damage dealt to be atleast 1, `player.health` can be lower and also throw an exception in the `courceIn` call.